python/ray/data/dataset.py

```diff
@@ -3394,6 +3394,13 @@ def write_bigquery(
                 overwritten if it exists.
             ray_remote_args: Kwargs passed to ray.remote in the write tasks.
         """
+        if ray_remote_args is None:
+            ray_remote_args = {}
+
+        # Each write task will launch individual remote tasks to write each block
+        # For correctness purposes, the write task cannot be retried
+        ray_remote_args["max_retries"] = 0
```

Comment:
Could we either not overwrite user-specified `max_retries`:

```suggestion
        if "max_retries" not in ray_remote_args:
            ray_remote_args["max_retries"] = 0
```

Or add a warning:
```suggestion
        if "max_retries" in ray_remote_args:
            warnings.warn(...)
        ray_remote_args["max_retries"] = 0
```

Reply:
I'll go with the latter suggestion since the first isn't very useful given that max_retries is always "specified" in https://github.com/ray-project/ray/blob/2c54abdb8aa0b3409667c7f4b1d06672d4b16b93/python/ray/data/_internal/remote_fn.py#L20-L28

Reply:
I think `ray_remote_args` takes precedence over the default configuration?

https://github.com/ray-project/ray/blob/8ecf2f5be0dfda4e2925da2ee1f08d7771e72e5e/python/ray/data/_internal/remote_fn.py#L29

Won't block on this, but I think we have the option to override if we want.

Reply:
Oh good catch. I'll raise a warning if we detect user specified max_retries > 0 and set it to 0 otherwise.